src/pyconverter/xml2py/utils.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (C) 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import pyconverter.xml2py.ast_tree as ast
+import yaml
+
+
+def parse_yaml(yaml_path):
+    """
+    Parse a YAML file.
+
+    Parameters
+    ----------
+
+    yaml_path : str
+        Path to the YAML file.
+    """
+    try:
+        with open(yaml_path, "r") as file:
+            yaml_data = yaml.safe_load(file)
+    except FileNotFoundError:
+        yaml_data = None
+    return yaml_data
+
+
+def get_config_data_value(yaml_path, value):
+    """
+    Return the value of a specific key in the YAML file.
+
+    Parameters
+    ----------
+
+    yaml_path : str
+        Path to the YAML file.
+
+    value : str
+        Key to search for in the YAML file.
+    """
+    config_data = parse_yaml(yaml_path)
+    try:
+        output = config_data[value]
+    except KeyError:
+        output = None
+
+    return output
+
+
+def create_name_map(meta_command, yaml_file_path):
+    # convert all to flat and determine number of occurances
+    proc_names = []
+    rules = get_config_data_value(yaml_file_path, "rules")
+    for cmd_name in meta_command:
+        cmd_name = cmd_name.lower()
+        if not cmd_name[0].isalnum():
+            cmd_name = cmd_name[1:]
```

Comment:
This condition is used twice in the ``utils.py`` code. The ``if`` section is indeed the same one but the ``else`` one is not.

l.71
```py
if not ans_name[0].isalnum():
    ans_name = ans_name[1:]
naive_names.append(ans_name)
```
l.84
```py
if not lower_name[0].isalnum():
    alpha_name = lower_name[1:]
else:
    alpha_name = lower_name
```

I don't feel it is necessary to create a function for this check as it would complexify the code in my opinion but feel free to reach out if you disagree with it.